10_predictive-modeling-review.qmd

```diff
@@ -0,0 +1,1045 @@
+       y = "Error") +
+  theme_minimal()
+```
+
+Our objective is to make accurate predictions on unseen data. It is important to make sure that models make accurate predictions on the modeling data and the implementation data.
+
+::: callout-tip
+## In-sample error
+
+The predictive error of a model measured on the data used to estimate the predictive model.
+:::
+
+::: callout-tip
+## Out-of-sample error
+
+The predictive error of a model measured on the data not used to estimate the predictive model. Out-of-sample error is generally greater than the in-sample error.
+:::
+
+::: callout-tip
+## Generalizability
+
+How well a model makes predictions on unseen data relative to how well it makes predictions on the data used to estimate the model.
+:::
+
+## Spending Data
+
+Predictive modelers strategically "spend data" to create accurate models that generalize to the implementation data. We will focus on two strategies: the training-testing split and v-fold cross validation.
+
+::: callout-tip
+## Training data
+
+A subset of data used to develop a predictive model. The share of data committed to a training set depends on the number of observations, the number of predictors in a model, and heterogeneity in the data. 0.8 is a common share.
+:::
+
+::: callout-tip
+## Testing data
+
+A subset of data used to estimate model performance. The testing set usually includes all observations not included in the testing set. Do not look at these data until the very end and only estimate the out-of-sample error rate on the testing set once. If the error rate is estimated more than once on the testing data, it will underestimate the error rate.
+:::
+
+![](images/cross-validation.jpeg){#fig-cv}
+
+::: callout-tip
+## $v$-fold Cross-Validation
+
+Break the data into $v$ equal-sized, exclusive groups. Train the model on data from $v - 1$ folds (analysis data) and measure the error metric (i.e. RMSE) on the left-out fold (assessment data). Repeat this process $v$ times, each time leaving out a different fold. Average the $v$ error metrics.
+:::
+
+Strategies for spending data differ based on the application and the size and nature of the data. Ideally, the training-testing split is made at the beginning of the modeling process. The $v$-fold cross-validation is used on the testing data for comparing
+
+1.  approaches for feature/target engineering
+2.  algorithms
+3.  hyperparameter tuning
+
+::: callout-tip
+## Data Leakage
+
+When information that won't be available when the model makes out-of-sample predictions is used when estimating a model. Looking at data from the testing set creates data leakage. Data leakage leads to an underestimate of out-of-sample error.
+:::
+
+::: {.callout-warning}
+Feature and target engineering is a common source of data leakage. For example, regularized regression models expect variables to be normalized (divide each predictor by the sample mean and divide by the sample standard deviation). 
```

Comment:
Also you don't define feature and target engineering and this is the first I believe you mention it. If you haven't defined them before in the course, this would be a good spot to do so.